src/fieldlayoutelements/BaseField.php

```diff
@@ -263,6 +266,59 @@ protected function id(): string
         return $this->attribute();
     }
 
+    /**
+     * Returns the `id` of the field instructions.
+     *
+     * @return string
+     * @since 3.7.24
+     */
+    protected function instructionsId(): string
+    {
+        return $this->id() . '-instructions';
+    }
+
+    /**
+     * Returns the `id` of the field tip.
+     *
+     * @return string
+     * @since 3.7.24
+     */
+    protected function tipId(): string
```

Comment:
How will tips and warnings be associated with particular input fields? Visually, it seems like they can be used to provide additional context about one or more fields at a time. Will we give users the option of what fields they will apply to?

Reply:
The IDs are gathered from `BaseField::describedBy()` here:

https://github.com/craftcms/cms/blob/00c77f7d8557687d4e028f793c10988504fc9b0d/src/fieldlayoutelements/BaseField.php#L311-L320

Which is called when setting `Field::$describedBy` before calling its `getInputHtml()` method:

https://github.com/craftcms/cms/blob/00c77f7d8557687d4e028f793c10988504fc9b0d/src/fieldlayoutelements/CustomField.php#L250

It’s up to the individual field types to actually pass their `describedBy` property to their input’s `aria-describedby` attribute. For example, Email fields do that by passing it to the `_includes/forms/text` template:

https://github.com/craftcms/cms/blob/00c77f7d8557687d4e028f793c10988504fc9b0d/src/fields/Email.php#L96

…which in turn will set it here:

https://github.com/craftcms/cms/blob/00c77f7d8557687d4e028f793c10988504fc9b0d/src/templates/_includes/forms/text.html#L33-L35